README.md
# Azure Logic Apps (Consumption) for Visual Studio Code

[![Version](https://img.shields.io/visual-studio-marketplace/v/ms-azuretools.vscode-logicapps.svg)](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-logicapps) [![Installs](https://img.shields.io/visual-studio-marketplace/i/ms-azuretools.vscode-logicapps.svg)](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-logicapps) [![Rating](https://img.shields.io/visual-studio-marketplace/r/ms-azuretools.vscode-logicapps.svg)](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-logicapps#review-details)

Azure Logic Apps (Consumption) simplifies how you build automated scalable workflows that integrate apps and data across cloud services and on-premises systems.

Use the Azure Logic Apps (Consumption) extension for VS Code to quickly create, debug, and manage Azure Logic Apps and Integration Accounts.

## Requirements

You must have Visual Studio Code 1.74.0 (November 2022) or later to install the extension.

All you need is an Azure Subscription to get started. If you don't have one, [click here](https://azure.microsoft.com/en-us/free/) for a free subscription.

## Features

* Browse, create, edit, and delete logic apps
* Browse and run logic apps triggers
* Browse and resubmit logic apps runs
* Browse and promote logic apps versions
* Browse, create, edit, and delete integration accounts, schema, map, partners, and agreements
* IntelliSense support helping you quickly and easily adding triggers, actions, and properties

* Create local logic apps project
* Add logic apps from explorer to local project
* Generate Azure resource manager deployment template for local project
* Generate Azure DevOps pipeline definition for local project

![Recurrence trigger IntelliSense](https://raw.githubusercontent.com/microsoft/vscode-azurelogicapps/main/resources/recurrence-trigger.gif)

![Run after IntelliSense](https://raw.githubusercontent.com/microsoft/vscode-azurelogicapps/main/resources/run-after.gif)

![Open in read-only designer](https://raw.githubusercontent.com/microsoft/vscode-azurelogicapps/main/resources/open-in-designer.gif)

![Open run in monitoring view](https://raw.githubusercontent.com/microsoft/vscode-azurelogicapps/main/resources/open-in-monitoring-view.gif)

## Recommended Extensions

We recommend installing the following VS Code extensions for a better experience when working with different integration account artifacts.

* For XSLT, install [XSLT Snippets](https://marketplace.visualstudio.com/items?itemName=marvinhagemeister.vscode-xslt-snippets)
* For Liquid, install [Shopify Liquid Template Snippets](https://marketplace.visualstudio.com/items?itemName=killalau.vscode-liquid-snippets)

## Managing Azure Subscriptions

If you are not signed in to Azure, you will see a "Sign in to Azure..." link. Alternatively, you can select "View->Command Palette" in the VS Code menu, and search for "Azure: Sign In".

If you don't have an Azure Account, you can sign up for one today for free and receive $200 in credits by selecting "Create a Free Azure Account..." or selecting "View->Command Palette" and searching for "Azure: Create an Account".

You may sign out of Azure by selecting "View->Command Palette" and searching for "Azure: Sign Out".

To select which subscriptions show up in the extension's explorer, click on the "Select Subscriptions..." button on any subscription node (indicated by a "filter" icon when you hover over it), or select "View->Command Palette" and search for "Azure: Select Subscriptions". Note that this selection affects all VS Code extensions that support the [Azure Account and Sign-In](https://github.com/Microsoft/vscode-azure-account) extension.

## Contributing

This project welcomes contributions and suggestions.  Most contributions require you to agree to a
Contributor License Agreement (CLA) declaring that you have the right to, and actually do, grant us
the rights to use your contribution. For details, visit https://cla.microsoft.com.

When you submit a pull request, a CLA-bot will automatically determine whether you need to provide
a CLA and decorate the PR appropriately (e.g., label, comment). Simply follow the instructions
provided by the bot. You will only need to do this once across all repos using our CLA.

This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or
contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

VS Code collects usage data and sends it to Microsoft to help improve our products and services. Read our [privacy statement](https://go.microsoft.com/fwlink/?LinkID=528096&clcid=0x409) to learn more. If you don’t wish to send usage data to Microsoft, you can set the `telemetry.enableTelemetry` setting to `false`. Learn more in our [FAQ](https://code.visualstudio.com/docs/supporting/faq#_how-to-disable-telemetry-reporting).


CHANGELOG.md
# Change Log
All notable changes to the "vscode-logicapps" extension will be documented in this file.

## [Unreleased]

## [1.2.10] - 2023-07-28
### Changed
- Update designer version to 2.30717.1.1

## [1.2.9] - 2023-07-17
### Changed
- Update designer version to 2.30701.1.2

## [1.2.8] - 2023-06-26
### Changed
- Update designer version to 2.30613.1.2

## [1.2.7] - 2023-06-09
### Changed
- Update designer version to 2.30529.1.7

## [1.2.6] - 2023-05-19
### Changed
- Update designer version to 2.30508.1.4

## [1.2.5] - 2023-05-12
### Changed
- Update designer version to 2.30501.1.5

## [1.2.4] - 2023-04-21
### Changed
- Update designer version to 2.30410.1.9

## [1.2.3] - 2023-04-07
### Changed
- Update designer version to 2.30322.1.6

## [1.2.2] - 2023-03-03
### Changed
- Update designer version to 2.30221.1.1

## [1.2.1] - 2023-02-10
### Changed
- Update designer version to 2.30131.1.1

## [1.2.0] - 2023-01-25
### Changed
- Update designer version to 2.30112.1.2

## [1.1.16] - 2022-12-02
### Changed
- Update designer version to 2.21110.1.5

## [1.1.15] - 2022-11-04
### Changed
- Update designer version to 2.21020.1.6

## [1.1.14] - 2022-10-14
### Changed
- Update designer version to 2.20929.1.4

## [1.1.13] - 2022-10-07
### Changed
- Update designer version to 2.20908.1.4

## [1.1.12] - 2022-09-09
### Changed
- Update designer version to 2.20825.1.2

## [1.1.11] - 2022-08-26
### Changed
- Update designer version to 2.20815.1.1

## [1.1.10] - 2022-08-12
### Changed
- Update designer version to 2.20731.1.2

## [1.1.9] - 2022-07-29
### Changed
- Update designer version to 2.20625.1.3

## [1.1.8] - 2022-07-01
### Changed
- Update designer version to 2.20606.1.3

## [1.1.7] - 2022-06-03
### Changed
- Update designer version to 2.20516.1.1

## [1.1.6] - 2022-05-05
### Changed
- Update designer version to 2.20418.1.2

## [1.1.5] - 2022-04-08
### Changed
- Update designer version to 2.20328.1.4

## [1.1.4] - 2022-03-24
### Changed
- Update designer version to 2.20314.1.3

## [1.1.3] - 2022-03-11
### Changed
- Update designer version to 2.20228.1.3

## [1.1.2] - 2022-02-25
### Changed
- Update designer version to 2.20214.1.1

## [1.1.1] - 2022-01-28
### Changed
- Update designer version to 2.20117.1.2

## [1.1.0] - 2022-01-13
### Added
- Add toggle to switch between old and new designer

### Changed
- Update designer version to 2.11227.1.2

## [1.0.26] - 2021-12-08
### Changed
- Update designer version to 2.11129.1.3

## [1.0.25] - 2021-11-16
### Changed
- Update designer version to 2.11108.1.1

## [1.0.24] - 2021-11-02
### Changed
- Update designer version to 2.11025.1.4

## [1.0.23] - 2021-10-19
### Changed
- Update designer version to 2.11011.1.5

## [1.0.22] - 2021-10-05
### Changed
- Update designer version to 2.10927.1.6

## [1.0.21] - 2021-09-21
### Changed
- Update designer version to 1.60913.1.4

## [1.0.20] - 2021-09-07
### Changed
- Update designer version to 1.60830.1.2
- Update logo icon

## [1.0.19] - 2021-08-25
### Changed
- Update designer version to 1.60816.1.2

## [1.0.18] - 2021-08-10
### Changed
- Update designer version to 1.60802.1.3

## [1.0.17] - 2021-07-27
### Changed
- Update designer version to 1.60719.1.6

## [1.0.16] - 2021-07-13
### Changed
- Update designer version to 1.60705.1.3

## [1.0.15] - 2021-07-02
### Changed
- Update designer version to 1.60621.1.4

## [1.0.14] - 2021-06-16
### Changed
- Update designer version to 1.60607.1.5

## [1.0.13] - 2021-06-01
### Changed
- Update designer version to 1.60524.1.5

## [1.0.12] - 2021-05-19
### Changed
- Update designer version to 1.60510.1.3

## [1.0.11] - 2021-05-04
### Changed
- Update designer version to 1.60426.1.4

## [1.0.10] - 2021-04-23
### Changed
- Update designer version to 1.60412.1.3

## [1.0.9] - 2021-04-08
### Changed
- Update designer version to 1.60329.1.1

## [1.0.8] - 2021-03-23
### Changed
- Update designer version to 1.60315.1.1

## [1.0.7] - 2021-03-09
### Changed
- Update designer version to 1.60301.1.3

## [1.0.6] - 2021-02-23
### Changed
- Update designer version to 1.60215.1.5

## [1.0.5] - 2021-02-09
Thanks, Tyler Leonhardt (@TylerLeonhardt), for the bug fix!

### Changed
- Update designer version to 1.60201.1.6

### Fixed
- #193 - Fix Draft.js script reference to work on case-sensitive file systems

## [1.0.4] - 2021-01-26
### Changed
- Update designer version to 1.60118.1.4

## [1.0.3] - 2021-01-19
### Changed
- Update dependencies

## [1.0.2] - 2021-01-12
### Changed
- Update designer version to 1.60104.1.1

## [1.0.1] - 2020-12-21
### Changed
- Update designer version to 1.51207.1.7

## [1.0.0] - 2020-12-03
### Changed
- Update designer version to 1.51123.1.5

## [0.2.60] - 2020-11-17
### Changed
- Update designer version to 1.51109.1.3

## [0.2.59] - 2020-11-12
### Changed
- Update designer version to 1.51102.1.1

## [0.2.58] - 2020-11-03
### Changed
- Update designer version to 1.51026.1.2

## [0.2.57] - 2020-10-27
### Changed
- Update designer version to 1.51019.1.4

## [0.2.56] - 2020-10-20
### Changed
- Update designer version to 1.51012.1.3

## [0.2.55] - 2020-10-12
### Changed
- Update designer version to 1.51005.1.1

## [0.2.54] - 2020-10-07
### Fixed
- #178 - Missing module

## [0.2.53] - 2020-10-05
### Changed
- Update designer version to 1.50928.1.2

## [0.2.52] - 2020-09-28
### Changed
- Update designer version to 1.50921.1.5

## [0.2.51] - 2020-09-21
### Changed
- Update designer version to 1.50914.1.5

## [0.2.50] - 2020-09-14
### Changed
- Update designer version to 1.50907.1.3

## [0.2.49] - 2020-09-09
### Changed
- Update designer version to 1.50831.1.1

## [0.2.48] - 2020-08-31
### Changed
- Update designer version to 1.50824.1.4

## [0.2.47] - 2020-08-24
### Changed
- Update designer version to 1.50817.1.1

## [0.2.46] - 2020-08-17
### Changed
- Update designer version to 1.50810.1.2

## [0.2.45] - 2020-08-10
### Changed
- Update designer version to 1.50803.1.1

## [0.2.44] - 2020-08-03
### Changed
- Update designer version to 1.50727.1.1

### Fixed

src/constants.ts
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See LICENSE in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

export class Constants {
    public static DesignerVersion = "2.30717.1.1";

    public static SubscriptionContextValue = "azureextensionui.azureSubscription";

    // #region Partners and Agreements

    public static Qualifier = new Map<string, string>([
        ["1 - D-U-N-S (Dun & Bradstreet)", "1"],
        ["2 - SCAC (Standard Carrier Alpha Code) (X12)", "2"],
        ["3 - Federal Maritime Commission", "3"],
        ["4 - IATA (International Air Transport Association)", "4"],
        ["7 - Global Location Number (GLN)", "7"],
        ["8 - UCC Communications ID (Uniform Code Council Communications)", "8"],
        ["9 - X.121 (CCITT) / D-U-N-S (Dun & Bradstreet) with 4 digit suffix", "9"],
        ["10 - Department of Defense", "10"],
        ["11 - Drug Enforcement Administration", "11"],
        ["12 - Telephone Number", "12"],
        ["13 - UCS Code", "13"],
        ["14 - D-U-N-S Plus Suffix / EAN (European Article Numbering Association)", "14"],
        ["15 - Petroleum Accountants Society of Canada Company Code", "15"],
        ["16 - D&B D-U-N-S Number plus 4-character suffix", "16"],
        ["17 - American Bankers Association", "17"],
        ["18 - Association of American Railroads (AAR) Standard Distribution Code / AIAG(Automotive Industry Action Group)", "18"],
        ["19 - EDI Council of Australia (EDICA) Communications ID Number (COMM ID)", "19"],
        ["20 - Health Industry Number", "20"],
        ["21 - Integrated Postsecondary Education Data System, or (IPEDS)", "21"],
        ["22 - Federal Interagency Commission on Education, or FICE / INSEE (Institut National de la Statistique et des Etudes Economiques) - SIREN", "22"],
        ["23 - National Center for Education Statistics Common Core of Data 12-Digit Number for Pre-K-Grade 12 Institutes, or NCES", "23"],
        ["24 - The College Board's Admission Testing Program 4-Digit Code of Postsecondary Institutes, or ATP", "24"],
        ["25 - ACT, Inc. 4-Digit Code of Postsecondary Institutions", "25"],
        ["26 - Statistics of Canada List of Postsecondary Institutions", "26"],
        ["27 - Health Care Financing Administration Carrier ID", "27"],
        ["28 - Health Care Financing Administration Fiscal Intermediary", "28"],
        ["29 - Health Care Financing Administration Medicare Provider", "29"],
        ["30 - U.S. Federal Tax Identification Number / ISO 6523: Organization Identification", "30"],
        ["31 - Jurisdiction Identification Number Plus 4 as assigned by the International Association of Industrial Accident Boards and Commissions (IAIABC)/ DIN (Deutsches Institut fuer Normung)", "31"],
        ["32 - U.S. Federal Employer Identification Number", "32"],
        ["33 - National Association of Insurance Commissioners Company Code (NAIC) / BfA (Bundesversicherungsanstalt fuer Angestellte)", "33"],
        ["34 - Medicaid Provider and Supplier Identification Number as assigned by individual State Medicaid Agencies in conjunction with Health Care Financing Administration (HCFA) / National Statistical Agency", "34"],
        ["35 - Statistics Canada Canadian College Student Information System Institution Codes", "35"],
        ["36 - Statistics Canada University Student Information System Institution Codes", "36"],
        ["37 - Society of Property Information Compilers and Analysts", "37"],
        ["38 - The College Board and ACT, Inc. 6-Digit Code List of Secondary Institutions", "38"],
        ["AM - Association Mexicana del Codigo de Producto (AMECOP) Communication ID", "AM"],
        ["NR - National Retail Merchants Association", "NR"],
        ["SA - User Identification Number as assigned by the Safety and Fitness Electronic Records (SAFER) System", "SA"],
        ["SN - Standard Address Number", "SN"],
        ["ZZ - Mutually Defined", "ZZ"],
        ["BT", "BT"],
        ["AS2Identity", "AS2Identity"],
        ["5 - INSEE (Institut National de la Statistique et des Etudes Economiques)", "5"],
        ["51 - GEIS (General Electric Information Services)", "51"],
        ["52 - INS (IBM Network Services)", "52"],
        ["53 - Datenzentrale des Einzelhandels", "53"],
        ["54 - Bundesverband der Deutschen Baustoffhaendler", "54"],
        ["55 - Bank Identifier Code", "55"],
        ["57 - KTNet (Korea Trade Network Services)", "57"],
        ["58 - UPU (Universal Postal Union)", "58"],
        ["59 - ODETTE (Organization for Data Exchange through Tele-Transmission in Europe)", "59"],
        ["61 - SCAC (Standard Carrier Alpha Code)", "61"],
        ["63 - ECA (Electronic Commerce Australia)", "63"],
        ["65 - TELEBOX 400 (Deutsche Telekom)", "65"],
        ["80 - NHS (National Health Service)", "80"],
        ["82 - Statens Teleforvaltning", "82"],
        ["84 - Athens Chamber of Commerce", "84"],
        ["85 - Swiss Chamber of Commerce", "85"],
        ["86 - US Council for International Business", "86"],
        ["87 - National Federation of Chambers of Commerce and Industry", "87"],
        ["89 - Association of British Chambers of Commerce", "89"],
        ["90 - SITA (Societe Internationale de Telecommunications)", "90"],
        ["91 - Assigned by seller or seller's agent", "91"],
        ["92 - Assigned by buyer or buyer's agent", "92"],
        ["103 - TW, Trade-van", "103"],
        ["128 - CH, BCNR (Swiss Clearing Bank Number)", "128"],
        ["129 - CH, BPI (Swiss Business Partner Identification)", "129"],
        ["144 - US, DoDAAC (Department of Defense Activity Address Code)", "144"],
        ["145 - FR, DGCP (Direction Generale de la Comptabilite Publique)", "145"],
        ["146 - FR, DGI (Direction Generale des Impots)", "146"],
        ["147 - JP, JIPDEC/ECPC (Japan Information Processing Development Corporation / Electronic Commerce Promotion Center)", "147"],
        ["148 - ITU (International Telecommunications Union) Data Network Identification Code (DNIC)", "148"],
        ["Z01 - Vehicle registration number", "Z01"],
        ["ZZZ - Mutually Defined", "ZZZ"]
    ]);

    public static AS2AllowedIdentityQualifiers = [
        "AS2Identity"
    ];

    public static X12AllowedIdentityQualifiers = [
        "1",
        "2",
        "3",
        "4",
        "7",
        "8",
        "9",
        "10",
        "11",
        "12",
        "13",
        "14",
        "15",
        "16",
        "17",
        "18",
        "19",
        "20",
        "21",
        "22",
        "23",
        "24",
        "25",
        "26",
        "27",
        "28",
        "29",
        "30",
        "31",
        "32",
        "33",
        "34",
        "35",
        "36",
        "37",
        "38",
        "AM",
        "NR",
        "SA",
        "SN",
        "ZZ",
        "BT"
    ];

    public static EdifactAllowedIdentityQualifiers = [
        "1",
        "4",
        "5",
        "8",
        "9",
        "12",
        "14",
        "18",
        "22",
        "30",
        "31",
        "33",
        "34",
        "51",
        "52",
        "53",
        "54",
        "55",
        "57",
        "58",
        "59",
        "61",
        "63",
        "65",
        "80",
        "82",
        "84",
        "85",
        "86",
        "87",
        "89",
        "90",
        "91",
        "92",
        "103",
        "128",
        "129",
        "144",
        "145",
        "146",
        "147",
        "148",
        "Z01",
        "ZZZ",
        "ZZ",
        "BT"
    ];

    // #endregion

    // Content Types
    public static JsonContentType = "application/json";
    public static LiquidContentType = "text/plain";
    public static TextContentType = "text/plain";
    public static XmlContentType = "application/xml";
}
